substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/substitute/UnsafeAutomaticSubstitutionProcessor.java

```diff
@@ -305,6 +304,12 @@ public void computeSubstitutions(ResolvedJavaType hostType, OptionValues options
         if (hostType.isArray()) {
             return;
         }
+
+        if (annotationSubstitutions.findSubstitution(hostType).isPresent()) {
```

Comment:
I'm curious: for which class does this trigger?

Reply:
A class within JBoss-Threads for one. 

We substitute the entire class, where the original has some static initializer we're scrubbing away and replacing.

Reply:
We already suppress all warnings by default for substituted types; see the end of the `reportUnsuccessfulAutomaticRecomputation()` method. But I would like to at least see a debug message about this decision, i.e., guarded by `Options.UnsafeAutomaticSubstitutionsLogLevel.getValue() >= DEBUG_LEVEL`.

Reply:
I think ultimately, if I recall, we replaced classed `Foo` entirely, but the old `Foo` ended up with a parsing error due to Unsafe substitutions that were being pointlessly done attempting to parse a class that referenced some unresolvable types or something. I forget the details at this point.

I'm not sure what the log message would be... `Didn't perform Unsafe substitutions on a class Foo that has been wholly replaced`?

Happy to add it, though.

Reply:
Just to reiterate: This case should already be fixed later on by:
https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/substitute/UnsafeAutomaticSubstitutionProcessor.java#L904
But if the substituted class references some missing types then you would end up with a parsing error as you mentioned. Bailing out early is better. The side effect is that you'll gate a "skipped unsafe substitution" message for every substituted type.